code/modules/biomatter_manipulation/solidifier.dm

```diff
@@ -0,0 +1,110 @@
+//Biomatter compressor
+//This machine converts liquid biomatter to solid one(sheets)
+//Working with this also required bio protection cloths
+
+#define BIOMATTER_PER_SHEET 		10
+#define CONTAINER_PIXEL_OFFSET 		6
+
+/obj/machinery/biomatter_solidifier
+	name = "Biomatter compressor"
+	icon = 'icons/obj/machines/simple_nt_machines.dmi'
+	icon_state = "solidifier"
+	density = TRUE
+	anchored = TRUE
+	use_power = 1
+	idle_power_usage = 5
+	active_power_usage = 300
+
+	circuit = /obj/item/weapon/circuitboard/neotheology/solidifier
+	var/active = FALSE
+	var/port_dir = NORTH
+	var/obj/structure/reagent_dispensers/biomatter/container
+
+
+/obj/machinery/biomatter_solidifier/update_icon()
+	if(active)
+		icon_state = "solidifier_on"
+	else
+		icon_state = initial(icon_state)
+
+
+/obj/machinery/biomatter_solidifier/Process()
+	if(active)
+		if(container)
+			if(container.reagents.has_reagent("biomatter", BIOMATTER_PER_SHEET))
+				container.reagents.remove_reagent("biomatter", BIOMATTER_PER_SHEET)
+
+				var/obj/item/stack/material/biomatter/current_stack
+				//if there any stacks here, let's check them
+				if(locate(/obj/item/stack/material/biomatter) in loc)
+					for(var/obj/item/stack/material/biomatter/stack_on_my_loc in loc)
+						//if this isn't full, we use that stack(current)
+						if(stack_on_my_loc.amount < stack_on_my_loc.max_amount)
+							current_stack = stack_on_my_loc
+							break
+
+				if(current_stack)
+					current_stack.amount += 1
+					current_stack.update_strings()
```

Comment:
That should be handled with at least `stack.add(AMOUNT)`

Reply:
With `stack.add(AMOUNT)` line
`current_stack.update_strings()`
is no longer required.